packages/alfa-aria/src/is-visible.ts

```diff
@@ -15,6 +15,15 @@ export function isVisible(node: Element | Text, context: Node): boolean {
       return false;
     }
 
+    let parent = getParentElement(node, context);
+    while (parent !== null) {
+      if (getAttribute(parent, "aria-hidden") === "true") {
+        return false;
+      }
+
+      parent = getParentElement(parent, context);
+    }
+
```

Comment:
The `getClosest()` function should be used in combination with the `flattened: true` option to ensure the function behaves correctly in Shadow DOM:

```ts
if (
  getClosest(
    node,
    context,
    node => isElement(node) && getAttribute(node, "aria-hidden") === "true",
    { flattened: true }
  ) !== null
) {
  return false;
}
```